Example.md

```diff
@@ -0,0 +1,977 @@
+For more information about GitHub REST API, please see https://developer.github.com/v3/activity/events/types/#pushevent
+
+Follow the these steps to create `cicd-event-listener`
+
+
+* Save the following content to `temp.yaml`
+
+```shell
+apiVersion: tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+  name: cicd-event-listener
+spec:
+  serviceAccountName: demo-sa
+  triggers:
+    - name: dev-ci-build-from-pr
+      interceptors:
+        - cel:
+            filter: (header.match("X-GitHub-Event", "pull_request") && body.action == "opened" || body.action == "synchronize") && body.pull_request.head.repo.full_name == "$GITHUB_USER/taxi"
+      bindings:
+        - name: dev-ci-build-from-pr-binding
+      template:
+        name: dev-ci-build-from-pr-template
+    - name: dev-cd-deploy-from-master
+      interceptors:
+        - cel:
+            filter: (header.match("X-GitHub-Event", "push") && body.repository.full_name == "$GITHUB_USER/taxi") && body.ref.startsWith("refs/heads/master")
+      bindings:
+        - name: dev-cd-deploy-from-master-binding
+      template:
+        name: dev-cd-deploy-from-master-template
+    - name: stage-ci-dryrun-from-pr
+      interceptors:
+        - cel:
+            filter: (header.match("X-GitHub-Event", "pull_request") && body.action == "opened" || body.action == "synchronize") && body.pull_request.head.repo.full_name == "$GITHUB_USER/taxi-stage-config"
+      bindings:
+        - name: stage-ci-dryrun-from-pr-binding
+      template:
+        name: stage-ci-dryrun-from-pr-template
+    - name: stage-cd-deploy-from-push
+      interceptors:
+        - cel:
+            filter: (header.match("X-GitHub-Event", "push") && body.repository.full_name == "$GITHUB_USER/taxi-stage-config") && body.ref.startsWith("refs/heads/master")
+      bindings:
+        - name: stage-cd-deploy-from-push-binding
+      template:
+        name: stage-cd-deploy-from-push-template
+
+```
+
+* Run the following command to replace strings and apply yaml
+
+**_NOTE:_**  Make sure `GITHUB_USER` envrionment variable is set.
+
+```shell
+cat temp.yaml  | sed s"/\$GITHUB_USER/$GITHUB_USER/" | oc apply -f -
+```
+
+## Expose webhook listener
+
+Create an OpenShift route to expose the EventLister Service
+
+### github-webhook-event-listener
+
+```shell
+cat <<EOF | oc apply -f -
+kind: Route
+apiVersion: route.openshift.io/v1
+metadata:
+  name: github-webhook-event-listener
+  labels:
+    app.kubernetes.io/managed-by: EventListener
+    app.kubernetes.io/part-of: Triggers
+    eventlistener: cicd-event-listener
+spec:
+  to:
+    kind: Service
+    name: el-cicd-event-listener
+    weight: 100
+  port:
+    targetPort: 8080
+  wildcardPolicy: None
+EOF
+```
+## Create github-auth secret
+
+Create secret with the github token that you have regenerated/downloaded.
+
+**_NOTE:_**  Make sure `GITHUB_USER` envrionment variable is set.
+```shell
+oc create secret generic ${GITHUB_USER}-github-auth --from-file="<path/to>/github-token.txt"
```

Comment:
You're creating this in a named environment, do you need a named github-auth token too?

Reply:
You don't have to have it.  I can remove it.